Modules/Microsoft.PowerShell.UnixCompleters/Microsoft.PowerShell.UnixCompleters/BashUtilCompleter.cs

```diff
@@ -0,0 +1,309 @@
+                .ToList();
+
+            completionResults.Sort(StringComparer.Ordinal);
+
+            string previousCompletion = null;
+            foreach (string completionResult in completionResults)
+            {
+                if (string.IsNullOrEmpty(completionResult))
+                {
+                    continue;
+                }
+
+                int equalsIndex = wordToComplete.IndexOf('=');
+
+                string completionText;
+                string listItemText;
+                if (equalsIndex >= 0)
+                {
+                    completionText = wordToComplete.Substring(0, equalsIndex) + completionResult;
+                    listItemText = completionResult;
+                }
+                else
+                {
+                    completionText = completionResult;
+                    listItemText = completionText;
+                }
+
+                if (completionText.Equals(previousCompletion))
+                {
+                    listItemText += " ";
+                }
+
+                previousCompletion = completionText;
+
+                yield return new CompletionResult(
+                    completionText,
+                    listItemText,
+                    CompletionResultType.ParameterName,
+                    completionText);
+            }
+        }
+
+        private string ResolveCommandCompleterFunction(string commandName)
+        {
+            if (string.IsNullOrEmpty(commandName))
+            {
+                throw new ArgumentException(nameof(commandName));
+            }
+
+            string completerFunction;
+            if (_commandCompletionFunctions.TryGetValue(commandName, out completerFunction))
+            {
+                return completerFunction;
+            }
+
+            string resolveCompleterInvocation = string.Format(s_resolveCompleterCommandTemplate, commandName);
+            completerFunction = InvokeBashWithArguments(resolveCompleterInvocation).Trim();
+            _commandCompletionFunctions[commandName] = completerFunction;
+
+            return completerFunction;
+        }
+
+        private string InvokeBashWithArguments(string argumentString)
+        {
+            using (var bashProc = new Process())
+            {
+                bashProc.StartInfo.FileName = this._bashPath;
+                bashProc.StartInfo.Arguments = argumentString;
+                bashProc.StartInfo.UseShellExecute = false;
+                bashProc.StartInfo.RedirectStandardOutput = true;
+                bashProc.Start();
+
+                return bashProc.StandardOutput.ReadToEnd();
+            }
+        }
+
+        private static string EscapeCompletionResult(string completionResult)
+        {
+            completionResult = completionResult.Trim();
+
+            if (!completionResult.Contains(' '))
+            {
+                return completionResult;
+            }
+
+            return "'" + completionResult.Replace("'", "''") + "'";
```

Comment:
I find this version simpler myself; we check to see if the case is simple, and if so return the simple answer. Otherwise, life is more complex and we do the complex thing. It aligns to my sense of laziness that I want to try the simple thing first. That way the complexity of the method increases as the context does, which makes sense to me